app/facades/trip_facade.rb

```diff
@@ -0,0 +1,17 @@
+class TripFacade
+  def self.get_trips(flight_params)
+    flights = FlightService.get_flights(flight_params)
+    trips = []
+    if flights[:message].nil?
+      flights[:data].each_with_index do |flight, index|
+        lat = flight[:attributes][:latitude]
+        lon = flight[:attributes][:longitude]
+        weather = WeatherService.get_weather(lat, lon)
+        trips << Trip.new(flight, weather, index)
+      end
+      trips #return value: an array of Trip Objects
+    else
+      flights
```

Comment:
Great job! I'm a bit confused here. What's the reason for returning `flights` here?

Reply:
The variable name `flights` is a bit misleading here. `flights` will either have a `:data` key with an array of flight hashes or a `:message` key containing an error. So if `flights[:message]` is nil we create `Trip` objects else we return the hash containing the error.

A better variable here would probably be `response`.

Reply:
OK, thank you for the clarification!